mne/viz/evoked.py

```diff
@@ -1472,7 +1472,11 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
     ch_types = list(set(channel_type(example.info, pick_)
                     for pick_ in picks))
     # XXX: could possibly be refactored; plot_joint is doing a similar thing
-    if any([type_ not in _DATA_CH_TYPES_SPLIT for type_ in ch_types]):
+    # Valid data types ordered for consistency (same as _plot_evoked)
+    valid_channel_types = ['eeg', 'grad', 'mag', 'seeg', 'eog', 'ecg', 'emg',
+                           'dipole', 'gof', 'bio', 'ecog', 'hbo', 'hbr',
+                           'misc']
```

Comment:
So which way we do take? 

From a design point of view I would use 'picks' or 'exclude' parameters to filter and let the user plot whatever they want. For instance, the default of 'exclude' could be 'valid' which means valid channel types (defined in io.pick) minus 'bads' channels to behave as it is now.